### PR TITLE
Make grafana deployment run as non-root and enable seccomp policy.

### PR DESCRIPTION
**Pull request by cjwagner**
/assign @mtaufen

### CODE REVIEW

File: config/prow/cluster/monitoring/grafana_deployment.yaml [hunk 1]
```diff
@@ -17,6 +19,8 @@ spec:
       app: grafana
   template:
     metadata:
+      annotations:
+        seccomp.security.alpha.kubernetes.io/pod: 'runtime/default'
```
Comment (mtaufen) [inline config/prow/cluster/monitoring/grafana_deployment.yaml:23]: It looks like this is actually available in securityContext now and the annotations are getting removed in 1.25. Let's try the securityContext version: https://kubernetes.io/docs/tutorials/clusters/seccomp/#create-a-pod-with-a-seccomp-profile-for-syscall-auditing

File: config/prow/cluster/monitoring/grafana_deployment.yaml [hunk 2]
```diff
@@ -30,6 +34,8 @@ spec:
         ports:
           - containerPort: 80
```
Comment (mtaufen) [inline config/prow/cluster/monitoring/grafana_deployment.yaml:35]: We might have to run with `CAP_NET_BIND_SERVICE` to bind low ports when running as non-root, depending on whether the Docker version is 1.19 or 1.20 (docker set `net.ipv4.ip_unprivileged_port_start=0` in 1.20, and I'm not sure what containerd does): https://github.com/moby/moby/pull/41030

Comment (cjwagner) [reply config/prow/cluster/monitoring/grafana_deployment.yaml:31]: I added a `capabilities` section to the pod's `securityContext` that I think should grant that capability.

**Review by cjwagner — COMMENTED** [review]
Also replaced the annotation with the `seccompProfile` field.

File: config/prow/cluster/monitoring/grafana_deployment.yaml (hunk 2 above)
Comment (stevekuznetsov) [reply config/prow/cluster/monitoring/grafana_deployment.yaml:31]: You could also just bind something higher and use the service to redirect it

**Comment by cjwagner** [comment]
I've switched the nginx proxy to serve on 8080 and redirect to 80 via the service as Steve suggested. PTAL.

**Comment by cjwagner** [comment]
The monitoring dashboard still seem to be working since this deployed. :+1:

**Pull request merged**